app/helpers/application_helper.rb

```diff
@@ -88,12 +88,138 @@ def donate_info
     @donate_info = { link: link, show: show }
   end
 
-  def election_event_title
-    "Wakefield and Tiverton & Honiton 2022 by-elections"
+  def by_election_constituencies
+    OnsConstituency.all.map(&:name)
+  end
+
+  def by_election_constituencies_ampersand
+    by_election_constituencies.map do |name|
+      name.gsub(" and ", " & ")
+    end
+  end
+
+  def by_election_constituencies_as_sentence
+    # "Wakefield and Tiverton & Honiton"
+    by_election_constituencies_ampersand.to_sentence
+  end
+
+  def election_event_title_with_year
+    if general_election?
+      "General Election #{election_year}"
+    else
+      "#{by_election_constituencies_as_sentence} #{election_year} by-elections"
+    end
+  end
+
+  def election_event_choice
+    if general_election?
+      "General Election"
+    else
+      # "Wakefield or Tiverton & Honiton by-elections"
+      options = { words_connector: ", ", last_word_connector: " or ", two_words_connector: " or " }
+      "#{by_election_constituencies_ampersand.to_sentence(options)} by-elections"
+    end
   end
 
   def election_hashtags
-    "#Wakefield or #TivertonandHoniton #byelection"
+    if general_election?
+      "#GeneralElection"
+    else
+      # "#Wakefield or #TivertonandHoniton #byelection"
+      hashtags = by_election_constituencies.map do |name|
+        "#" + name.gsub(/[^A-Za-z]/, "")
+      end
+      options = { words_connector: ", ", last_word_connector: " or ", two_words_connector: " or " }
+      "#{hashtags.to_sentence(options)} #byelection"
+    end
+  end
+
+  def election_date
+    date_from_env = ENV["ELECTION_DATE"]
+    return Date.parse(date_from_env) unless date_from_env.nil? || (date_from_env == "")
+    Date.parse("2022-6-23")
```

Comment:
Looks like some debugging which got left in?

Reply:
No, that was meant to be a default, for any environment (e.g. heroku previews ?) where we forget to set ELECTION_DATE .

Reply:
Ah I see.  I would have a very mild preference for it failing to start up if the variable was missing, as this would alert us to the incorrect configuration rather than upgrading Heroku workers to something with the wrong date... but it's a tiny corner case and really doesn't matter :-)